src/frontend/src/sol/services/worker.sol-wallet.services.ts

```
@@ -0,0 +1,95 @@
+import {
+	solAddressDevnetStore,
+	solAddressLocalnetStore,
+	solAddressMainnetStore,
+	solAddressTestnetStore
+} from '$lib/stores/address.store';
+import type { WalletWorker } from '$lib/types/listener';
+import type {
+	PostMessage,
+	PostMessageDataRequestSol,
+	PostMessageDataResponseError
+} from '$lib/types/post-message';
+import type { Token } from '$lib/types/token';
+import {
+	isNetworkIdSOLDevnet,
+	isNetworkIdSOLLocal,
+	isNetworkIdSOLTestnet
+} from '$lib/utils/network.utils';
+import { syncWallet, syncWalletError } from '$sol/services/sol-listener.services';
+import type { SolPostMessageDataResponseWallet } from '$sol/types/sol-post-message';
+import { mapNetworkIdToNetwork } from '$sol/utils/network.utils';
+import { assertNonNullish } from '@dfinity/utils';
+import { get } from 'svelte/store';
+
+export const initSolWalletWorker = async ({ token }: { token: Token }): Promise<WalletWorker> => {
+	const {
+		id: tokenId,
+		network: { id: networkId }
+	} = token;
+
+	const WalletWorker = await import('$sol/workers/sol-wallet.worker?worker');
+	const worker: Worker = new WalletWorker.default();
+
+	const isTestnetNetwork = isNetworkIdSOLTestnet(networkId);
+	const isDevnetNetwork = isNetworkIdSOLDevnet(networkId);
+	const isLocalNetwork = isNetworkIdSOLLocal(networkId);
+
+	worker.onmessage = ({ data }: MessageEvent<PostMessage<SolPostMessageDataResponseWallet>>) => {
+		const { msg } = data;
+
+		switch (msg) {
+			case 'syncSolWallet':
+				syncWallet({
+					tokenId,
+					data: data.data as SolPostMessageDataResponseWallet
+				});
+				return;
+
+			case 'syncSolWalletError':
+				syncWalletError({
+					tokenId,
+					error: (data.data as PostMessageDataResponseError).error,
+					hideToast: isTestnetNetwork || isDevnetNetwork || isLocalNetwork
```

Can you add a `TODO` to review this at some point. As far as I remember neither @DenysKarmazynDFINITY nor myself were super happy with this pattern `hideToast`. Right Denys?

---

Yep, we wanted to revisit the approach at some point (e.g. by avoiding requests (or launching wallets in this case) that we know in advance are not gonna succeed, rather than making a call + failing silently).